fe/fe-core/src/main/java/com/starrocks/metric/ResourceGroupMetricMgr.java

```diff
@@ -0,0 +1,232 @@
+// This file is made available under Elastic License 2.0.
+// This file is based on code available under the Apache license here:
+//   https://github.com/apache/incubator-doris/blob/master/fe/fe-core/src/main/java/org/apache/doris/metric/MetricRepo.java
+
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package com.starrocks.metric;
+
+import com.starrocks.catalog.WorkGroup;
+import com.starrocks.qe.ConnectContext;
+import com.starrocks.qe.QueryDetail;
+import com.starrocks.qe.SessionVariable;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class ResourceGroupMetricMgr {
+    private static final Logger LOG = LogManager.getLogger(ResourceGroupMetricMgr.class);
+
+    private static final String QUERY_RESOURCE_GROUP = "query_resource_group";
+    private static final String QUERY_RESOURCE_GROUP_LATENCY = "query_resource_group_latency";
+    private static final String QUERY_RESOURCE_GROUP_ERR = "query_resource_group_err";
+
+    private static final ConcurrentHashMap<String, LongCounterMetric> RESOURCE_GROUP_QUERY_COUNTER_MAP
+            = new ConcurrentHashMap<>();
+    private static final ConcurrentHashMap<String, List<GaugeMetricImpl>> RESOURCE_GROUP_QUERY_LATENCY_MAP
+            = new ConcurrentHashMap<>();
+    private static final ConcurrentHashMap<String, LongCounterMetric> RESOURCE_GROUP_QUERY_ERR_COUNTER_MAP
+            = new ConcurrentHashMap<>();
+
+    //starrocks_fe_query_resource_group
+    public static void increaseQuery(ConnectContext ctx, Long num) {
+        SessionVariable sessionVariable = ctx.getSessionVariable();
+        if (!sessionVariable.isEnableResourceGroup()) {
+            return;
+        }
+        WorkGroup workGroup = ctx.getWorkGroup();
+        if (workGroup == null) {
```

Comment:
What's the case?

Reply:
For example enable_pipeline_engine=false, or sql does not hit the resource group in the fe service.